### PR TITLE
Added data warnings on selected datasets

### PR DESCRIPTION
**Pull request by nerik**
https://globalfishingwatch.atlassian.net/browse/MAP-702https://globalfishingwatch.atlassian.net/browse/MAP-702

Screenshot 2022-03-01 at 10 28 17

Here shown with fallback text, waiting for datasets i18n

### CODE REVIEW

File: libs/i18n-labels/source/datasets.json [hunk 1]
```diff
@@ -1034,6 +1034,8 @@
   "public-eez-land": {
     "name": "EEZ (marineregions.org)",
     "description": "Flanders Marine Institute (2019). Maritime Boundaries Geodatabase: Maritime Boundaries and Exclusive Economic Zones (200NM), version 11. Source: marineregions.org",
+    "dataWarning": "This platform uses EEZ reference layers from Marine Regions",
```
Comment (j8seangel) [inline libs/i18n-labels/source/datasets.json:1037]: this file is going to be overwritten by the [script that autogenerates](https://github.com/GlobalFishingWatch/crowdin-datasets-internationalization) this file based on the api datasets status (basically: config.translate = true).
Now it runs manually and we could keep it by hand but it is going to be automate soon so we need to:
- include these properties on the dataset configuration
or
- move these to the source translations app scope and ensure the id matches using [this config](https://github.com/GlobalFishingWatch/frontend/pull/1268/files#diff-6ac73904c9a979f3e0150f088b8639608618633f9a93689539e065bdfd3bfb27R26)

File: apps/fishing-map/features/workspace/context-areas/ContextAreaLayerPanel.tsx [hunk 2]
```diff
@@ -111,6 +118,35 @@ function LayerPanel({ dataview, onToggle }: LayerPanelProps): React.ReactElement
           {isUserLayer && <Remove dataview={dataview} />}
         </div>
       </div>
+      {layerActive && DATASOURCE_WARNING.includes(dataset?.id) && (
+        <div className={cx(styles.properties, styles.dataWarning)}>
+          <div>
+            {t(
+              `datasets:${dataset?.id}.dataWarning` as any,
```
Comment (j8seangel) [inline apps/fishing-map/features/workspace/context-areas/ContextAreaLayerPanel.tsx:125]: depending on the decision at https://github.com/GlobalFishingWatch/frontend/pull/1268/files#diff-f50769714a5978c2c452a0e4491daa04fcd316493eabf1c60a9272640a206735R1037 we'll need to keep datasets or change to the default i18n namespace

File: apps/fishing-map/features/datasets/datasets.hook.ts [hunk 3]
```diff
@@ -197,3 +199,23 @@ export const useAutoRefreshImportingDataset = (
     }
   }, [dataset, dispatchFetchDataset, refreshTimeout])
 }
+
+export const useAddDataset = ({datasetCategory, onSelect}: NewDatasetTooltipProps) => {
+  const { dispatchDatasetModal, dispatchDatasetCategory } = useDatasetModalConnect()
+  return async () => {
```
Comment (j8seangel) [inline apps/fishing-map/features/datasets/datasets.hook.ts:205]: async is not needed anymore but thanks for making it reusable

**Review by j8seangel — CHANGES_REQUESTED** [review]
We need to make a decision on the warning translations place, please take a look at the comments

File: apps/fishing-map/features/workspace/context-areas/ContextAreaLayerPanel.tsx [hunk 4]
```diff
@@ -139,9 +140,11 @@ function LayerPanel({ dataview, onToggle }: LayerPanelProps): React.ReactElement
               onClose={onDataWarningModalClose}
               contentClassName={styles.modalContent}
             >
-              {t(
-                `datasets:${dataset?.id}.dataWarningDetail` as any,
-                'This platform uses a reference layer from an external source.'
+              {ReactHtmlParser(
+                t(
+                  `datasets.datasetSpecific.${dataset?.id}.dataWarningDetail` as any,
```
Comment (nerik) [inline apps/fishing-map/features/workspace/context-areas/ContextAreaLayerPanel.tsx]: Already fixed to `dataset.datasetSpecific`

**Pull request merged**